service/src/main/java/org/apache/griffin/core/job/JobServiceImpl.java

```diff
@@ -666,9 +668,31 @@ public void triggerJobById(Long id) throws SchedulerException {
                     .forJob(jobKey)
                     .startNow()
                     .build();
-            scheduler.scheduleJob(trigger);
+            TriggerKey key = trigger.getKey();
+            CountDownLatch latch = new CountDownLatch(1);
+            String listenerName = "listenerJob_" + jobKey.toString();
+            try {
+                scheduler.getListenerManager().addTriggerListener(
+                        new CountDownTriggerListener(latch, listenerName)
+                        , key::equals);
+
+                scheduler.scheduleJob(trigger);
+
+                try {
+                    latch.await(0, TimeUnit.MILLISECONDS);
```

Comment:
why timeout is not used here?

Reply:
Timeout don't used here by mistake, I fixed it